.circleci/config.yml

```diff
@@ -1158,6 +1158,7 @@ jobs:
   t_ubu_ubsan_clang_cli:
     <<: *base_ubuntu2004_clang_small
```

Comment:
From what I can tell, this step inherits from `base_ubuntu2004_clang_small` and then overrides the resource class with `medium`. `base_ubuntu2004_clang_small` is however only used once in this file, so it might make sense to rename it to `base_ubuntu2004_clang_medium` and set the resource class to `medium` in it directly.
It would be nice if @r0qs or @cameel were to take a look, since I'm far from a devops/circleci expert.

Reply:
Added these changes in the latest push. 

Should we go ahead and increase `MAKEFLAGS: -j 2` to a higher number ?  I noticed that compared the resource class `large` seems to have a higher allowance for the `-j` flag.

Reply:
Looks like `small` resource classes usually have `-j2` set, and `large` have `-j5`, so I'd go with `-j3` here - it's default for the base `base_ubuntu2004_clang` anyway, so technically you don't even need to specify it (even though it would be nice for readability).

Reply:
Ah, yes. Just noticed that the entire `environment` section is redundant. Will remove all of it.